src/controllers/AuthController.ts

```diff
@@ -0,0 +1,33 @@
+import bcrypt from 'bcryptjs'
+import environment from '../config/environment'
+import { sign, Secret } from 'jsonwebtoken'
+import { Request, Response } from 'express'
+import { User } from '../models'
+import validation from '../validation/'
+import ISignup from '../interfaces'
+
+export default class AuthController {
+  public secretKey : Secret
+
+  constructor () {
+    this.secretKey = environment.secretKey
+  }
+
+  public static async creation ({ name, email, role, password, res }:ISignup) {
```

Comment:
i think this function is not necessary for this context, 
it must be controller function have request and response,
u can include this operation in your sign up chain operation
@salsabeelomar 
what is your opinion @aaamra ?